src/project.cc

```diff
@@ -311,9 +313,22 @@ Project::Entry GetCompilationEntryFromCompileCommandEntry(
     compiler_driver = cleanup_maybe_relative_path(compiler_driver);
   result.args.push_back(compiler_driver);
 
-  // Add -working-directory if not provided.
-  if (!AnyStartsWith(args, "-working-directory"))
-    result.args.emplace_back("-working-directory=" + entry.directory);
+  optional<CompilerType> compiler_type;
+
+  if (g_disable_compiler_type_extraction_for_test) {
```

Comment:
What about making `g_disable_compiler_type_extraction_for_test` an `optional<CompilerType>`? Then the test can specify which compiler driver it wants to use.

Reply:
Updated as you suggested.